src/main/java/de/dennisguse/opentracks/services/handlers/TrackPointCreator.java

```diff
@@ -114,16 +115,28 @@ public void stop() {
         this.context = null;
     }
 
+    public void onChange(@NonNull Location location) {
+        synchronized (this) {
+            TrackPoint trackPoint = new TrackPoint(location, createNow());
+            addSensorData(trackPoint);
+            onNewTrackPoint(trackPoint);
```

Comment:
First `addSensorData` then `onNewTrackPoint`. In `onNewTrackPoint` line 139 the `addSensorData` is called again. Is that correct?

Reply:
Nice catch; I removed the call - would just be wasting time.